portfolio/src/main/java/com/google/sps/servlets/LikeCommentServlet.java

```diff
@@ -37,29 +39,34 @@
     public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
         long id = Long.parseLong(request.getParameter("id"));
 
-        //get comment entity corresponding to id from datastore
-        Key commentEntityKey = KeyFactory.createKey("Comment", id);
         DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+        Transaction txn = datastore.beginTransaction();
 
         try {
+            //get comment entity corresponding to id from datastore
+            Key commentEntityKey = KeyFactory.createKey("Comment", id);
             Entity commentEntity = datastore.get(commentEntityKey);
-
+            
             //calculate the number of likes using the previous number of likes
             long numLikes = (long) commentEntity.getProperty("numLikes") + 1;
 
             commentEntity.setProperty("numLikes", numLikes);
 
             datastore.put(commentEntity);
 
-            saveLikedComment(datastore, id);
-
+            txn.commit();
+             saveLikedComment(datastore, id);
```

Comment:
You'll want to do this operation in the same transaction. You can pass a reference the txn to the `saveLikedComment` method. Otherwise even if the transaction fails, you'll still save the "like" without incrementing the counter and you'll end up with an inconsistent state.

Reply:
For some reason, when I called `saveLikedComment` in the same transaction, it kept giving me an `IllegalArgumentException`, and when I moved it outside it seemed to work again. Any ideas why this would happen?

Reply:
hard to know without seeing the full stacktrace, but my guess is you're not passing the txn as an arg. When you call `datastore.put` you'll need the `txn` as the first argument.

Reply:
I tried that and I still get the same error: 
```
java.lang.IllegalArgumentException: cross-group transaction need to be explicitly specified, see TransactionOptions.Builder.withXGfound both Element {|  type: "Comment"|  id: 0x11c00000000000|}| and Element {|  type: "Like"|  name: "1858047
64220139124118"|}|
```

Should I create a new transaction to deal with the `Like` entity?

Reply:
I see that you've bypassed the issue by eliminating the second operation, but the hint here is `cross-group transaction need to be explicitly specified, see TransactionOptions.Builder.withXG`

This doc has some more info on cross-group transactions:
https://cloud.google.com/appengine/docs/standard/java/datastore/transactions#using_cross-group_transactions